src/mem/Allocator.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 1996-2022 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_SRC_MEM_ALLOCATOR_H
+#define SQUID_SRC_MEM_ALLOCATOR_H
+
+#include "mem/forward.h"
+
+namespace Mem
+{
+
+/// An interface for memory allocators that deal with fixed-size objects.
+/// Allocators may optimize repeated de/allocations using memory pools.
+class Allocator
+{
+public:
+    Allocator(char const *aLabel) : label(aLabel) {}
```

Comment:
If possible, please make this single-parameter constructor explicit:

```suggestion
    explicit Allocator(const char * const aLabel): label(aLabel) {}
```